### PR TITLE
[SIEM] [Docs] Detection engine rules API

### LINKED ISSUE #702: [Docs] Document detection engine rules API
Document the rules API, and the JSON schema of rules and signals.

### PR DESCRIPTION
**Pull request by benskelker**
Start documenting the rules API, as specified in [#455](https://github.com/elastic/siem-team/issues/455) and #702.

[Preview](http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/detection-engine-overview.html)

Resolves #702

### CODE REVIEW

**Comment by benskelker** [comment]
Hi @FrankHassanabad 

This is an ongoing draft. I've left you lots of questions. We've got time, so just get back to me when you've got the time. Thanks!

**Comment by benskelker** [comment]
Hi @FrankHassanabad and @dhurley14 

I think this is ready for a full review. Can one of you give it a look?
Thanks (and thanks Frank for all your time and help while I was working on the docs)

**Comment by benskelker** [comment]
Hi @jmikell821 and @DonNateR,

Could one of you review this? 

Thanks

**Comment by DonNateR** [comment]
@benskelker: I'll take it.

File: docs/en/siem/index-api-overview.asciidoc [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+=== Index endpoint
+
+You use the index endpoint to create, get, and delete a signal index in a 
+{kib} space.
+
+You can only create a signal index when the user role has `manage` privileges 
+for both the {ref}/security-privileges.html[{es} cluster and the signals index].
+
+NOTE: To make sure administrators can always create indices, in the `Indices` 
+field on the *Create role* page use a glob pattern to match indices from
```
Comment (DonNateR) [inline docs/en/siem/index-api-overview.asciidoc]: Add a comma after `page,`

File: docs/en/siem/index-api-overview.asciidoc [hunk 2]
```diff
@@ -0,0 +1,107 @@
+[[index-api-overview]]
+=== Index endpoint
+
+You use the index endpoint to create, get, and delete a signal index in a 
+{kib} space.
+
+You can only create a signal index when the user role has `manage` privileges 
+for both the {ref}/security-privileges.html[{es} cluster and the signals index].
```
Comment (DonNateR) [inline docs/en/siem/index-api-overview.asciidoc]: Is the "signals index" the name of SIEM or backend element? If so, keep the sentence as is, but if there's some noun confusion there. In its current usage, the word "signals" is genitive, implying that the signals posses the index. Switching to "signal index" would help clear up some of the confusion.

File: docs/en/siem/privileges-api-overview.asciidoc [hunk 3]
```diff
@@ -0,0 +1,97 @@
+[[privileges-api-overview]]
+[role="xpack"]
+=== Privileges endpoint
+
+Retrieves the user's {kib} space and index privileges, which determine if the  
+user can create an index for the SIEM signals generated by detection engine 
+rules. To create the index, users require `manage` privileges for both the 
+{ref}/security-privileges.html[{es} cluster and the signals index].
```
Comment (DonNateR) [inline docs/en/siem/privileges-api-overview.asciidoc]: Same comment as the one I made on the "index-api-overview.asciidoc" file.

File: docs/en/siem/rules-api-bulk-actions.asciidoc [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+// KIBANA
+
+===== Response code
+
+`200`:: 
+    Indicates a successful call.
+    
+===== Response payload
+
+A JSON array of that includes a unique ID for each rule. A unique rule ID is
```
Comment (DonNateR) [inline docs/en/siem/rules-api-bulk-actions.asciidoc]: Remove "of".

File: docs/en/siem/rules-api-overview.asciidoc [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+== Detection engine API
+
+You can create rules that automatically turn events and alerts sent to the
+{siem-app} into signals. These signals are displayed on the Engine Detection 
+page. 
+
+For more information on signals, and the difference between signals, events, 
+and alerts, see <<det-engine-terminology, detection engine terminology>>.
+
+The RESTful API has these endpoints:
```
Comment (DonNateR) [inline docs/en/siem/rules-api-overview.asciidoc]: I would remove "RESTful". More of a marketing buzzy term and the API is already explained API endpoint descriptions below.

File: docs/en/siem/rules-ui-create.asciidoc [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+.. _Risk score_: A numerical value between 0 and 100 that correlates with the _Severity_ level. General guidelines are:
+... `21` represents low severity.
+... `47` represents medium severity.
+... `73` represents high severity.
+... `99` represents critical severity.
+.. _Reference URLs_ (optional): References to information that is relevant to 
+the rule. For example, links to the relevant Mitre URL.
+.. _False positives_ (optional): List of common scenarios that may produce 
+false-positive signals.
+.. _Tags_ (optional): Words and phrases that can be used to categorize, filter,
```
Comment (DonNateR) [inline docs/en/siem/rules-ui-create.asciidoc]: Change from passive to active, "Words and phrases used to categorize, filter, and search the rule".

File: docs/en/siem/rules-ui-create.asciidoc [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+documents added to indices during the last 6 minutes. This overlap helps ensure 
+signals are issued even when incoming host and network data is delayed.
+. Save the rule with or without activation.
+
+[float]
+=== Manage existing rules
+
+. Go to *SIEM* -> *Detection Engine*.
+. Do one of the following:
+* Click the all actions icon (three dots), and select the required action.
```
Comment (DonNateR) [inline docs/en/siem/rules-ui-create.asciidoc]: Remove the comma after "(three dots)".

**Comment by benskelker** [comment]
Thanks @DonNateR - great feedback.
Sorry. I should have mentioned it in the initial request, but only the API needs reviewing. I'm still working on the other stuff.

**Comment by DonNateR** [comment]
Got it @benskelker, I'll update my review!

File: docs/en/siem/detection-engine-intro.asciidoc [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+The detection engine automatically searches for threats and creates signals 
+when they are detected. These threats are detected using rules that define the 
+conditions required for a signal to be issued. The {siem-app} comes 
+preconfigured with rules that search for suspicious activity on your network 
+and hosts. Additionally, to meet your specific requirements,
+you can <<rules-ui-create, create your own rules>>.
+
+[float]
+=== Signals
+*Ben: check with Garrett*
```
[the remark this answers is not in the document]
Comment (benskelker) [reply docs/en/siem/detection-engine-intro.asciidoc]: Removed text and added placeholder comment

File: docs/en/siem/detection-engine-intro.asciidoc [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+When a rule's criteria are matched, a signal is produced and displayed on the
```
[the remark this answers is not in the document]
Comment (benskelker) [reply docs/en/siem/detection-engine-intro.asciidoc]: Removed text and added placeholder comment

File: docs/en/siem/detection-engine-intro.asciidoc [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+To see whether a rule is active, when it last ran, and how far back it searched 
+for signals during its last execution, view the Activity monitor table
+(*Manage rules* -> *Activity monitor*).
+
+[float]
+[[det-engine-terminology]]
+=== Terminology
+
+Signals::
+Always refer to a {siem-soln} produced detection. That is, signals are never
```
[the remark this answers is not in the document]
Comment (benskelker) [reply docs/en/siem/detection-engine-intro.asciidoc]: Removed text and added placeholder comment

**Comment by benskelker** [comment]
retest

**Comment by benskelker** [comment]
run elasticsearch-ci/docs

**Comment by dhurley14** [comment]
Just a general question - is it possible to add screenshots to the docs? Screenshots might help contextualize details as to where buttons are etc.

File: docs/en/siem/rules-api-create.asciidoc [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+      }
+     }
+  ],
+  "enabled": false
+}
+--------------------------------------------------
+// KIBANA
+
+<1> The rule runs every hour.
+<2> When the rule runs it analyzes data from 70 minutes before its start time.
```
Comment (dhurley14) [inline docs/en/siem/rules-api-create.asciidoc]: Should we leave a link to the date math docs here? https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#date-math

Comment (benskelker) [reply docs/en/siem/rules-api-create.asciidoc]: @dhurley14 
This page includes the JSON rule schema table, which contains a link to the date maths docs: http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/rules-api-create.html

Here's a [link](http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/rule-api-overview.html) to a preview of the docs in the first comment, which displays how they'll appear when they are published.

File: docs/en/siem/rules-api-get.asciidoc [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+[source,js]
+--------------------------------------------------
+GET api/detection_engine/rules?id=16947168-5405-453d-a8b5-aadad357af42
+--------------------------------------------------
+// KIBANA
+
+==== Response code
+
+`200`:: 
+    Indicates a successful call.
```
Comment (dhurley14) [inline docs/en/siem/rules-api-get.asciidoc]: Can we add an example response here? Should look like this

```json
{
  "created_at": "2020-01-28T16:34:57.923Z",
  "updated_at": "2020-01-28T16:36:44.233Z",
  "created_by": "elastic_kibana",
  "description": "Query with a rule_id that acts like an external id",
  "enabled": true,
  "false_positives": [],
  "from": "now-6m",
  "id": "9d82f1eb-7777-8888-aaaa-d15d7194c68c",
  "immutable": false,
  "interval": "30s",
  "rule_id": "query-rule-id",
  "language": "kuery",
  "output_index": ".siem-signals-devin-hurley-default",
  "max_signals": 100,
  "risk_score": 1,
  "name": "Changes only the name to this new value",
  "query": "user.name: root or user.name: admin",
  "references": [],
  "severity": "high",
  "updated_by": "user",
  "tags": [],
  "to": "now",
  "type": "query",
  "threat": [],
  "version": 3,
  "status": "succeeded",
  "status_date": "2020-01-28T16:37:48.396Z",
  "last_success_at": "2020-01-28T16:37:48.396Z",
  "last_success_message": "succeeded"
}
```

File: docs/en/siem/rules-api-find.asciidoc [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+[source,js]
+--------------------------------------------------
+GET api/detection_engine/rules/_find?page=1&per_page=5&sort_field=enabled&sort_order=asc&filter=alert.attributes.name:windows
+--------------------------------------------------
+// KIBANA
+
+==== Response code
+
+`200`:: 
+    Indicates a successful call.
```
Comment (dhurley14) [inline docs/en/siem/rules-api-find.asciidoc]: sample response if you want to include.

```json
{
  "page": 1,
  "perPage": 20,
  "total": 1,
  "data": [
    {
      "created_at": "2020-01-29T23:10:13.950Z",
      "updated_at": "2020-01-29T23:10:14.441Z",
      "created_by": "elastic_kibana",
      "description": "Query with a rule_id that acts like an external id",
      "enabled": true,
      "false_positives": [],
      "from": "now-6m",
      "id": "ca9f25aa-ffff-4444-3333-1ec301a02529",
      "immutable": false,
      "interval": "5m",
      "rule_id": "query-rule-id",
      "language": "kuery",
      "output_index": ".siem-signals-devin-hurley-default",
      "max_signals": 100,
      "risk_score": 1,
      "name": "Query with a rule id",
      "query": "user.name: root or user.name: admin",
      "references": [],
      "severity": "high",
      "updated_by": "user",
      "tags": [],
      "to": "now",
      "type": "query",
      "threat": [],
      "version": 1,
      "status": "succeeded",
      "status_date": "2020-01-29T23:10:19.904Z",
      "last_success_at": "2020-01-29T23:10:19.904Z",
      "last_success_message": "succeeded"
    }
  ]
}
```

File: docs/en/siem/rules-api-delete.asciidoc [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+[source,js]
+--------------------------------------------------
+DELETE api/detection_engine/rules?id=16947168-5405-453d-a8b5-aadad357af42
+--------------------------------------------------
+// KIBANA
+
+==== Response code
+
+`200`:: 
+    Indicates a successful call.
```
Comment (dhurley14) [inline docs/en/siem/rules-api-delete.asciidoc]: Sample response for delete. Not sure if we want to include a sample response for delete.

```json
{
  "created_at": "2020-01-29T23:10:13.950Z",
  "updated_at": "2020-01-29T23:12:56.282Z",
  "created_by": "user",
  "description": "Query with a rule_id that acts like an external id",
  "enabled": true,
  "false_positives": [],
  "from": "now-6m",
  "id": "ca9f25aa-ffff-5555-ffff-1ec301a02529",
  "immutable": false,
  "interval": "5m",
  "rule_id": "query-rule-id",
  "language": "kuery",
  "output_index": ".siem-signals-devin-hurley-default",
  "max_signals": 100,
  "risk_score": 1,
  "name": "Changes only the name to this new value",
  "query": "user.name: root or user.name: admin",
  "references": [],
  "severity": "high",
  "updated_by": "user",
  "tags": [],
  "to": "now",
  "type": "query",
  "threat": [],
  "version": 2,
  "status": "succeeded",
  "status_date": "2020-01-29T23:10:19.904Z",
  "last_success_at": "2020-01-29T23:10:19.904Z",
  "last_success_message": "succeeded"
}
```

**Comment by FrankHassanabad** [comment]
We have these endpoints if you think we should document them. They are pre-packaged rules endpoints. I didn't see them in the docs.
---
* <kibana URL>/api/detection_engine/prepackaged
* <kibana URL>/api/detection_engine/prepackaged/_status

Here are the scripts which you can see the output of them:
---
* detection_engine/scripts/add_prepackaged_rules.sh
* detection_engine/scripts/get_prepackaged_rules_status.sh

These output status of the pre_packged rules as well as being able to add them
without using the UI if the user wants to. 


Examples:
```ts
./get_prepackaged_rules_status.sh 
{
  "rules_custom_installed": 0,
  "rules_installed": 0,
  "rules_not_installed": 143,
  "rules_not_updated": 0
}
```

```ts
./add_prepackaged_rules.sh 
{
  "rules_installed": 143,
  "rules_updated": 0
}
```

File: docs/en/siem/rules-api-overview.asciidoc [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+
+Token-based authentication, using the same username and password used to log in
+to the Kibana UI, is required to access the API.
+
+[float]
+=== API calls
+
+All calls to the API are stateless. Each call must include all the information {siem-soln} needs to perform the request.
+
+All requests must contain these headers:
```
Comment (FrankHassanabad) [inline docs/en/siem/rules-api-overview.asciidoc]: Technically, only POST and PUT calls require these whenever you have JSON that is part of the body. In the scripts for example you have to use this for a `post_rule.sh` or you will get back an error that it requires something in the body.

However for a `get_rule_by_id.sh` where it does a GET request it doesn't need it because it is not pushing any data to the server.

If for some reason your POST/PUT REST call doesn't have a body you can also omit this header. It doesn't hurt to have it though and a lot of times we just copy and paste scripts with it for GET calls when the server doesn't care because we rarely have a body with the GET.

So, in short, this is only required if the GET/POST/PUT happens to have a body that is a JSON, otherwise the content-type is not needed.

**Comment by FrankHassanabad** [comment]
Just my opinion but here:

Screen Shot 2020-01-29 at 5 39 22 PM

I would show a screen shot of that in KQL like so:

Screen Shot 2020-01-29 at 5 41 38 PM

And take a moment to expound a bit about the Index patterns where for this type of rule you would want to focus it only on indexes which contain windows events such as `winlogbeat-*` for performance reasons.

Also, I would point them to the name of the pre-packaged rule they can duplicate and modify for the example where this came from such as:

`Process started by MS Office program - possible payload`

Just asking because I was wondering looking at it if there was an already created rule or something like it I could use to connect from the docs to the product.

**Comment by FrankHassanabad** [comment]
Per @randomuserid I think we need to adjust wording to this:

```
Critical: Signals that indicate a security incident has occurred.
```

to just have something like:

```
Critical: Signals that indicate it is very likely a security incident could have occurred.
```

Just lighten it up slightly to indicate it is very likely but false positives are always possible.

**Comment by FrankHassanabad** [comment]
I think for this:
Screen Shot 2020-01-29 at 6 04 16 PM


Ranges might be better?

```
0 - 21 represents low severity
22 - 47 represents medium severity
48 - 73 represents high severity
74 - 100 represents critical severity
```

It feels a bit weird to see 99 but wonder well where's 100 ;-)

You can set it to 100. We don't set any of the pre-packaged rules to 100 (FYI) because we kind of allow the user to set any custom rules to 100 so they know if they ever see 100 it is there rule they created completely and not us.

**Comment by FrankHassanabad** [comment]
My opinion on this one but it's really maybe a nit.

For tags:

Screen Shot 2020-01-29 at 6 08 44 PM

We might want to let them know that we use `Elastic` for all the pre-built rules at this point and they should avoid using it. We don't "block" them from adding it though if they want to.

Screen Shot 2020-01-29 at 6 08 37 PM

I am sure that should be obvious to them, but pointing it out in case you think it should be spelled out for them. We also use a few other tags such as `network`, and `linux` and they are fine to use any of those as they want to help categorize their rules.

**Comment by FrankHassanabad** [comment]
Nice! 👍 
Screen Shot 2020-01-29 at 6 13 03 PM

That is really concise and perfect way of saying that very few words 🎉 

You don't have to make changes here but the only other thing I can think of adding would be something to answer the question of, "Well when would the user increase this number and why?" They should increase the number if they have a rule which is not very performant or they have a very low/default setting for `max_workers` in their Kibana yml file:

```ts
xpack.task_manager.max_workers: 10
```

In combination with a lot of rules turned on. I don't know where `task_manager.max_workers` is documented at the moment btw.

If the user begins to see errors in their failure tab about "gap detection issues" ... 

Screen Shot 2020-01-29 at 6 29 02 PM

These are errors that the rule could not run within a given that given 1 minute of overlap. This is a tricky one to document because of our value of "It depends"
https://www.elastic.co/about/our-source-code#it-depends

which is pretty true here. It depends a lot on how many rules they enable, how many Kibana instances they are running, hardware, etc... If they keep increasing the number then at some point it will hurt rather than hurt them as it will just slow down the queries even further.

But I know they are going to ask us for advice when they see these gap issues and the few remedies to help would be:

* Add another Kibana instance
* Increase slowly the max_workers
* And finally, maybe increase for a few particularly slow rules the look behind by more than the default 1 minute.

It could really be any of those combinations that help them out. Hard to say. Up to you on how to document this (if at all), I just want to point these things out.
  
We didn't explicitly tell the user that they should increase their `max_workers` btw in that error message because the big con of increasing `max_workers` could be that more workers are suddenly putting a strain on the system with SIEM signals and suddenly they experience other issues.

**Comment by FrankHassanabad** [comment]
Screen Shot 2020-01-29 at 6 38 05 PM

Technically, this runs _close_ to 5 minutes. If the max_workers is set to say 10 and is very busy it will run maybe a bit past 5 minutes. If your kibana instance is shut down for a long time and restarted _all_ the 5 minute rules will be past their time to run and will all execute immediately and be "clumped" together forever every 5 minutes.

No changes I think, but just want to point out that it is close but not exact which is why there is the additional look back time to help compensate for the fuzziness.

**Comment by FrankHassanabad** [comment]
For these two looking closer:

Screen Shot 2020-01-29 at 6 42 14 PM

/privileges returns the user's permissions for the Kibana Space, the signals index, and if the user is an authenticated user or not.

/signals - Aggregates and returns signals, and updates their statuses, and also it can do normal queries. See `./signals/query_signals.sh` for details on it being able to do queries as well.

**Comment by FrankHassanabad** [comment]
Screen Shot 2020-01-29 at 6 47 33 PM

Per earlier conversations I would change this to not imply they need `Content-Type: application/json` when it is a GET call

For the example I would either use a POST/PUT example if you want to show they need it for those calls or drop it from the `/rules/_find` example as it is not needed there.

**Comment by FrankHassanabad** [comment]
This is not required, but rather optional for query based rules:
Screen Shot 2020-01-29 at 6 50 15 PM

Really there is no longer a `filter based rules` but only `query` or `saved_query` types now:
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/schemas.ts#L51

In either situation it is optional now.

**Comment by FrankHassanabad** [comment]
Screen Shot 2020-01-29 at 6 55 13 PM

Index is optional and the default if it is not provided will actually be dynamically set at execution time to whatever the index patterns in siem advanced settings are for the index. We however, for pre-packaged rules and in the UI always set this setting at rule creation time.

We might make it required in a future release though if we end up not using the feature. That part is still up for debate since we haven't used the feature on the front end fwiw.

**Comment by FrankHassanabad** [comment]
As per earlier comment for the API I would suggest using ranges maybe for risk scores in the docs and make 99 go to 100:

Screen Shot 2020-01-29 at 6 59 42 PM

**Comment by FrankHassanabad** [comment]
Looking through:
http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/rules-api-create.html

There is one new option of `version` that will be tricky to document:

https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/schemas.ts#L105

This is not required but can be set at the API layer when creating a rule, and the number can also be changed with the update API. If not set at rule creation time it will default to `1`. If not set in an update it will auto-increment by 1 for each update and all updates (except if the only update is that the rule is enabled or disabled in which case it will not update its version number).

For pre-packaged rules the version number will start at 1 and _never_ increment unless we ship new rules with a new version. In that case we increment the pre-pacakged rules our selves and the user will see this in their UI when the number is increased:

Screen Shot 2020-01-29 at 7 05 40 PM

If you want to see this happen yourself in your environment choose a pre-packaged rule such as this one:
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/rules/prepackaged_rules/windows_command_shell_started_by_svchost.json#L35
 
And change the version from a 1 to 2, and restart Kibana and visit the manage API rules page.

**Comment by FrankHassanabad** [comment]
nit:

I would encourage the user to use `70m` rather than seconds to match your wording. The UI is interesting in that it chooses to turn things into seconds for its convience but really we should encourage users to use sensible minutes for most of the time:
Screen Shot 2020-01-29 at 7 09 50 PM

No changes needed, just a nit.

**Comment by FrankHassanabad** [comment]
Aww you're the best @benskelker ! You have version there. That's right 👍 
Screen Shot 2020-01-29 at 7 12 26 PM

**Comment by FrankHassanabad** [comment]
Yes! Thank you thank you! Sorry for the tech debt and leaky abstractions but thank you so much for the great NOTE! Puts me in a good mood. Really appreciate this:

Screen Shot 2020-01-29 at 7 14 06 PM

For background info, we are going to try to talk more with Kibana core team members and see what we can do for breaking changes in version 8.0.0 with regards to mixed camel and snake case stuff and how we can normalize it all.

**Comment by FrankHassanabad** [comment]
For all the updates here:
http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/rules-api-update.html

Each one of those fields are optional. Not a single one is required. Also none of those have defaults either. I don't know if you copied that over from the create section but we need that re-written to basically just enumerate them all as optional.

I would add a note that `version` will not update if they are only updating `enabled` from true to false or false to true.

This part is perfect!
Screen Shot 2020-01-29 at 7 24 10 PM

I think the docs are good at indicating the user can pass in either an id or rule_id but not both.

**Comment by FrankHassanabad** [comment]
Another good tricky one.

For this part of the docs:
http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/index-api-overview.html

We probably should document the ILM policy we create when we make the index and bootstrap with.

Here is the IML policy we create:
https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/index/signals_policy.json

We name the policy and alias with the same name as the index. This should make the user understand we make more than the index, we actually create a bootstrap index and a ILM policy with it and what that is by default.

**Comment by FrankHassanabad** [comment]
For exports:
http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/rules-api-export.html

Screen Shot 2020-01-29 at 7 38 53 PM

I would add that the default if this is not specified is that it will export everything.

**Comment by FrankHassanabad** [comment]
For this part:
http://stack-docs_699.docs-preview.app.elstc.co/guide/en/siem/guide/master/privileges-api-overview.html

There is an addition of:

```ts
"is_authenticated": true
```

That @XavierM added for us where if the user is not an authenticated user (Elastic without user logins enabled) then that will be false, otherwise it will be true.

Example of it right now:

```ts
{
  "username": "elastic_kibana",
  "has_all_requested": false,
  "cluster": {
    "monitor_ml": true,
    "manage_ccr": true,
    "manage_index_templates": true,
    "monitor_watcher": true,
    "monitor_transform": true,
    "read_ilm": true,
    "manage_api_key": true,
    "manage_security": true,
    "manage_own_api_key": false,
    "manage_saml": true,
    "all": true,
    "manage_ilm": true,
    "manage_ingest_pipelines": true,
    "read_ccr": true,
    "manage_rollup": true,
    "monitor": true,
    "manage_watcher": true,
    "manage": true,
    "manage_transform": true,
    "manage_token": true,
    "manage_ml": true,
    "manage_pipeline": true,
    "monitor_rollup": true,
    "transport_client": true,
    "create_snapshot": true
  },
  "index": {
    ".siem-signals-hassanabad-frank-default": {
      "all": true,
      "manage_ilm": true,
      "read": true,
      "create_index": true,
      "read_cross_cluster": true,
      "index": true,
      "monitor": true,
      "delete": true,
      "manage": true,
      "delete_index": true,
      "create_doc": true,
      "view_index_metadata": true,
      "create": true,
      "manage_follow_index": true,
      "manage_leader_index": true,
      "write": true
    }
  },
  "application": {},
  "is_authenticated": true
}
```

**Comment by benskelker** [comment]
> Looks great! I appreciated seeing the detections terminology piece. I think that will be very helpful going forward.

@dhurley14 - thanks for the feedback

**Comment by benskelker** [comment]
@FrankHassanabad @dhurley14 
Thanks very much for the reviews. I've corrected the docs. Don't know if you want to have another look before I merge the PR.

Frank - I'll fix the comments on the UI docs in a separate PR.

**Comment by benskelker** [comment]
@FrankHassanabad 
This should fix everything.

@spong - I don't know if you want to take another look, but I added stuff to the UI sections about the required encryption key and missing signals.

Thanks again - you've really helped me and that makes for better docs.
